cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java

```diff
@@ -185,6 +188,84 @@ private String createInput(String folderName) throws IOException {
     return readResponse(response, new TypeToken<List<ScheduledRuntime>>() { }.getType());
   }
 
+  private String getLocalDatasetPath(ProgramId workflowId, String runId) {
+    String path = String.format("apps/%s/workflows/%s/runs/%s/localdatasets", workflowId.getApplication(),
+                                workflowId.getProgram(), runId);
+
+    return getVersionedAPIPath(path, Constants.Gateway.API_VERSION_3_TOKEN, workflowId.getNamespace());
+  }
+
+  private Map<String, DatasetCreationSpec> getWorkflowLocalDatasets(ProgramId workflowId, String runId)
+    throws Exception {
+    HttpResponse response = doGet(getLocalDatasetPath(workflowId, runId));
+    return readResponse(response, new TypeToken<Map<String, DatasetCreationSpec>>() { }.getType());
+  }
+
+  private void deleteWorkflowLocalDatasets(ProgramId workflowId, String runId) throws Exception {
+    doDelete(getLocalDatasetPath(workflowId, runId));
+  }
+
+  @Test
+  public void testLocalDatasetDeletion() throws Exception {
+    HttpResponse response = deploy(WorkflowAppWithLocalDatasets.class,
+                                   Constants.Gateway.API_VERSION_3_TOKEN, TEST_NAMESPACE2);
+    Assert.assertEquals(200, response.getStatusLine().getStatusCode());
+
+    File waitFile = new File(tmpFolder.newFolder() + "/wait.file");
+    File doneFile = new File(tmpFolder.newFolder() + "/done.file");
+
+    ProgramId workflowId = new ProgramId(TEST_NAMESPACE2, WorkflowAppWithLocalDatasets.NAME, ProgramType.WORKFLOW,
+                                         WorkflowAppWithLocalDatasets.WORKFLOW_NAME);
+
+    startProgram(workflowId.toId(), ImmutableMap.of("wait.file", waitFile.getAbsolutePath(),
+                                                    "done.file", doneFile.getAbsolutePath(),
+                                                    "dataset.*.keep.local", "true"));
+
+    while (!waitFile.exists()) {
+      TimeUnit.MILLISECONDS.sleep(50);
+    }
+
+    String runId = getRunIdOfRunningProgram(workflowId.toId());
+
+    doneFile.createNewFile();
+
+    waitState(workflowId.toId(), ProgramStatus.STOPPED.name());
+
+    Map<String, DatasetCreationSpec> localDatasets = getWorkflowLocalDatasets(workflowId, runId);
+    Assert.assertEquals(2, localDatasets.size());
+    Assert.assertEquals("MyTable", localDatasets.get("MyTable." + runId).getInstanceName());
```

Comment:
Yes. If we know the actual instance name(MyTable.runId), then it can be used with the regular dataset service endpoints, so we do not need the endpoints for individual local datasets.